quinn/src/endpoint.rs

```diff
@@ -413,14 +461,14 @@ impl State {
                                 buf,
                                 &mut response_buffer,
                             ) {
-                                Some(DatagramEvent::NewConnection(handle, conn)) => {
-                                    let conn = self.connections.insert(
-                                        handle,
-                                        conn,
-                                        self.socket.clone(),
-                                        self.runtime.clone(),
-                                    );
-                                    self.incoming.push_back(conn);
+                                Some(DatagramEvent::NewConnection(incoming)) => {
+                                    if self.incoming.len() < MAX_INCOMING_CONNECTIONS {
```

Comment:
Does it still make sense to check against `MAX_INCOMING_CONNECTIONS` here when we now allow the caller to decide whether to accept the connection?

Reply:
Yeah. `MAX_INCOMING_CONNECTIONS` is a new thing we're introducing in this PR. When the endpoint driver task receives a potentially connection-creating packet, it pushes an `Incoming` to `quinn::endpoint::State`'s `incoming` `VecDeque`, from which the user the application can get it through `Endpoint::accept`. The size of this queue should be close to zero in healthy conditions. We're adding this conservative limit so that an endpoint doesn't just use indefinitely more memory in a situation where incoming connection attempts are happening faster than the application is processing them for a prolonged period of time.